File: Front/eventos.client/src/app/models/evento.ts
import { Lote } from "./lote";
import { Palestrante } from "./palestrante";
import { RedeSocial } from "./rede-social";

export interface Evento {
  id: number; 
  nome: string; 
  local: string; 
  dataEvento:Date;
  tema: string;
  qtdPessoas: number;
  imagemURL: string;
  dataCadastro: Date;
  email: string;
  telefone: string;

  //FK
  lotes: Lote[];
  redesSociais: RedeSocial[]; 
  palestrantesEventos: Palestrante[];  
     

}

File: Back/Eventos.Domain/Models/Evento.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;


namespace Eventos.Domain.Models;

//[Table("EventoNovoNome")]
public class Evento
{
    //[Key]
    public int Id { get; set; }
    [Required]
    public string? Nome { get; set; }
    [Required]
    public string? Local { get; set; }
    [Required]
    public DateTime? DataEvento { get; set; }
    public string? Tema { get; set; }
    public int QtdPessoas { get; set; }

    public string ImagemURL { get; set; } = string.Empty;
    public DateTime DataCadastro { get; set; }

    //[NotMapped] - nâo leva ao banco esse campo

    public string Email { get; set; } = string.Empty;

    public string Telefone { get; set; } = string.Empty;

    public IEnumerable<Lote> Lotes { get; set; } 
    public IEnumerable<RedeSocial> RedesSociais { get; set; } 


    public IEnumerable<PalestranteEvento> PalestrantesEventos { get; set; }



}
